testsuite/testsuite.c

```diff
@@ -554,14 +681,104 @@ void join_thread(THREAD_TYPE thread)
 #elif defined(NETOS)
     /* TODO: */
 #else
-    int res = WaitForSingleObject((HANDLE)thread, INFINITE);
+    DWORD res = WaitForSingleObject((HANDLE)thread, INFINITE);
     assert(res == WAIT_OBJECT_0);
     res = CloseHandle((HANDLE)thread);
     assert(res);
     (void)res; /* Suppress un-used variable warning */
 #endif
 }
 
+#endif /* SINGLE_THREADED */
+
+#ifndef NO_FILESYSTEM
+
+#ifdef _MSC_VER
+#include <direct.h>
+#endif
+
+#define TMP_DIR_PREFIX "tmpDir-"
+/* len is length of tmpDir name, assuming
+ * len does not include null terminating character */
+char* create_tmp_dir(char *tmpDir, int len)
+{
+    if (len < (int)XSTR_SIZEOF(TMP_DIR_PREFIX))
+        return NULL;
+
+    XMEMCPY(tmpDir, TMP_DIR_PREFIX, XSTR_SIZEOF(TMP_DIR_PREFIX));
+
+    if (mymktemp(tmpDir, len, len - XSTR_SIZEOF(TMP_DIR_PREFIX)) == NULL)
+        return NULL;
+
+#ifdef _MSC_VER
+    if (_mkdir(tmpDir) != 0)
+        return NULL;
+#else
+    if (mkdir(tmpDir, 0700) != 0)
+        return NULL;
+#endif
+
+    return tmpDir;
+}
+
+int rem_dir(const char* dirName)
+{
+#ifdef _MSC_VER
+    if (_rmdir(dirName) != 0)
+        return -1;
+#else
+    if (rmdir(dirName) != 0)
+        return -1;
+#endif
+    return 0;
+}
+
+int rem_file(const char* fileName)
+{
+#ifdef _MSC_VER
+    if (_unlink(fileName) != 0)
+        return -1;
+#else
+    if (unlink(fileName) != 0)
+        return -1;
+#endif
+    return 0;
+}
+
+int copy_file(const char* in, const char* out)
+{
+    byte buf[2500];
```

Comment:
Not a fan of hard coded 2500 size. 

I think it is to accommodate these:
```
ls -la certs/crl/crl.pem
-rw-r--r--  1 davidgarske  staff  2229 Feb  1 09:03 certs/crl/crl.pem
davidgarske@Davids-MBP-2 wolfssl % ls -la certs/crl/crl.revoked
-rw-r--r--  1 davidgarske  staff  2330 Feb  1 09:03 certs/crl/crl.revoked
```

Why not just use heap?

Reply:
There is no cross-platform way to check the file size (that I could find). I'll see if there is a better way to do this.

Reply:
Changed to reading/writing in 100 byte chunks. That should be fine for stack size.